app/models/system_discussion_post.rb

```diff
@@ -0,0 +1,13 @@
+class SystemDiscussionPost
```

Comment:
Can you change this to be a service (`CreateSystemDiscussionPost`) like the other ones that already exist pls (although it doesn't have to inherit etc like the others).